core/src/main/java/org/wildfly/extras/creaper/core/online/SslOptions.java

```diff
@@ -0,0 +1,232 @@
+        } catch (GeneralSecurityException ex) {
+            throw new IllegalArgumentException("Failed to init KeyManagerFactory for " + this.keyStorePath, ex);
+        }
+
+        return keyManagerFactory.getKeyManagers();
+    }
+
+    private TrustManager[] getTrustManagers() {
+        if (this.trustStorePath == null) {
+            return null;
+        }
+
+        final TrustManagerFactory trustManagerFactory = createTrustManagerFactory();
+        final KeyStore trustStore = createStore(this.trustStoreType);
+        final FileInputStream trustStoreStream = createStoreStream(this.trustStorePath);
+
+        try {
+            trustStore.load(trustStoreStream, toCharArray(this.trustStorePassword));
+        } catch (IOException ex) {
+            throw new IllegalStateException("Failed to load truststore " + this.trustStorePath, ex);
+        } catch (GeneralSecurityException ex) {
+            throw new IllegalStateException("Failed to load truststore " + this.trustStorePath, ex);
+        } finally {
+            closeQuietly(trustStoreStream);
+        }
+
+        try {
+            trustManagerFactory.init(trustStore);
+        } catch (KeyStoreException ex) {
+            throw new IllegalArgumentException("Failed to init TrustManagerFactory for " + this.trustStorePath, ex);
+        }
+        return trustManagerFactory.getTrustManagers();
+    }
+
+    private static KeyManagerFactory createKeyManagerFactory() {
+        try {
+            return KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        } catch (NoSuchAlgorithmException ex) {
+            throw new IllegalStateException("KeyManagerFactory cannot be instantiated with default algorithm.", ex);
+        }
+    }
+
+    private static TrustManagerFactory createTrustManagerFactory() {
+        try {
+            return TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        } catch (NoSuchAlgorithmException ex) {
+            throw new IllegalStateException("TrustManagerFactory cannot be instantiated with default algorithm.", ex);
+        }
+    }
+
+    private static KeyStore createStore(String storeType) {
+        try {
+            return KeyStore.getInstance(storeType);
+        } catch (KeyStoreException ex) {
+            throw new IllegalArgumentException("KeyStore of type '" + storeType + "' cannot be instantiated.", ex);
+        }
+    }
+
+    private static FileInputStream createStoreStream(String storePath) {
+        try {
+            return new FileInputStream(storePath);
+        } catch (FileNotFoundException ex) {
+            throw new IllegalArgumentException("There is no store file " + storePath, ex);
+        }
+    }
+
+    private static char[] toCharArray(String password) {
+        if (password != null && !password.isEmpty()) {
+            return password.toCharArray();
+        }
+
+        return null;
+    }
+
+    private static void closeQuietly(FileInputStream stream) {
```

Comment:
BTW, you can use Guava if you wish. Creaper Core already has a dependency on it.